### ISSUE TITLE
Implement event / listener instead of model observers

### ISSUE THREAD

**Issue by thijsvdanker** [hyn-me/multi-tenant#13]
I'm trying to get my head around what's happening in the code, and came across the Observers.
Looking for some documentation on the pattern I found that what's called Model Observers in Laravel 4.1 (http://laravel.com/docs/4.2/eloquent#model-observers) is now described as Listeners (http://laravel.com/docs/5.1/events#defining-listeners) and Eloquent Events (http://laravel.com/docs/5.1/eloquent#events).

Would it make sense to rewrite the observers into the event / listener setup?
I think it's practically the same, but as it would follow the L5.1 standards it might be more easy to recognise what's going on?

**Comment by Luceos** [comment]
Yes that would make perfect sense. My professional efforts are mostly in a laravel app of version 4.2; so I know most of the patterns used there and did not take the time to investigate the changes for that pattern in L5 and up. Eloquent events are less documented than the previous observers, which is a shame.

In fact the api documentation still [refers](http://laravel.com/api/5.1/Illuminate/Database/Eloquent/Model.html#method_observe) to observe, observing and observers; so perhaps this would be a task when the method observe is renamed or removed from the model class.

The actual modification would only affect the namespace and class name of the observers, so nothing serious.

So yes and no; not critical, but if touching that code, why not :+1:
